sustainability_purchase_stock/tests/test_purchase.py

```diff
@@ -0,0 +1,74 @@
+from odoo import Command
+
+from odoo.addons.sustainability_purchase.tests.common import CarbonPurchaseCommon
+
+
+class TestSustainabilityPurchaseStock(CarbonPurchaseCommon):
+    def test_delivery_address_field(self):
+        """
+        Check that sustainability_partner_shipping_id field works correctly
+        """
+        self.env["res.partner"].create(
+            {
+                "parent_id": self.purchase_partner_1.id,
+                "type": "delivery",
+                "name": "Delivery Contact",
+            }
+        )
+        po = (
+            self.env["purchase.order"]
+            .with_context(force_onchange=True)
+            .create(
+                {
+                    "partner_id": self.purchase_partner_1.id,
+                }
+            )
+        )
+        po._onchange_partner_id_set_sustainability_shipping()
+        addresses = self.purchase_partner_1.address_get(["delivery"])
+        delivery_id = addresses.get("delivery")
+        expected_id = delivery_id or self.purchase_partner_1.id
```

Comment:
You can do it like that if you want (no needs, only for knowledge)
```suggestion
        expected_id = addresses.get("delivery", self.purchase_partner_1.id)
```